src/controllers/user.controller.ts

```diff
@@ -103,14 +108,51 @@ export default class UserController {
 
     static async saveUser(ctx: Koa.Context) {
         const userRepo: Repository<User> = getRepository(User)
-        const user: any = JWT.verify(ctx.params.token, process.env.JWT_SECRET, function (err: any, decoded: User): User {
-            return decoded
-        });
+        let newUser: User = await userRepo.findOne({
+            where: {
+                activationLink: ctx.params.link
+            }
+        })
+        if (!newUser) ctx.throw(HttpStatus.NOT_FOUND, 'User not found')
+        newUser.isActivated = true
+        const tokens: {accessToken: string, refreshToken: string} = generateTokens({id: newUser.id})
+        newUser.refreshToken = tokens.refreshToken
+        ctx.cookies.set('refreshToken',  tokens.refreshToken, {maxAge: 30 * 24 * 60 * 60 * 1000, httpOnly: true})
+        await userRepo.save(newUser)
+        ctx.redirect(`${process.env.CLIENT_URL}login`)
+        ctx.body = {
+            data: newUser
+        }
+    }
+
+    static async logout(ctx: Koa.Context){
+        const userRepo: Repository<User> = getRepository(User)
+        let user: User = await userRepo.findOne(ctx.params.user_id)
+        if (!user) ctx.throw(HttpStatus.NOT_FOUND, "User not found")
+        user.refreshToken = null
+        ctx.cookies.set('refreshToken', null)
         await userRepo.save(user)
         ctx.redirect(`${process.env.CLIENT_URL}login`)
         ctx.body = {
-            data: user
+            user
```

Comment:
Seems a little strange, because in the line above, the format is as follows (line 142):
```
        ctx.body = {
            data: newUser
        }
```